### PR TITLE
Fix mersi 2 angles reading

### PR DESCRIPTION
**Pull request by ninahakansson**
The MERSI-2 reader could no longer read angles data due to that keyword 'calibration' was always expected.

The error message:
```
python3.6/site-packages/satpy/readers/mersi2_l1b.py", line 95, in get_dataset
    if dataset_id['calibration'] == 'counts':
KeyError: 'calibration'
```

 - [x] Tests added 
 - [x] Passes ``flake8 satpy``

### CODE REVIEW

File: satpy/readers/mersi2_l1b.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def get_dataset(self, dataset_id, ds_info):
 
         fill_value = attrs.pop('FillValue', np.nan)  # covered by valid_range
         valid_range = attrs.pop('valid_range', None)
-        if dataset_id['calibration'] == 'counts':
+        if 'calibration' in dataset_id and dataset_id['calibration'] == 'counts':
```
Comment (mraspaud) [inline satpy/readers/mersi2_l1b.py:95]: The more pythonic way to do this is:
```suggestion
        if dataset_id.get('calibration') == 'counts':
```
IIRC this should work with DataIDs too.

**Review by mraspaud — COMMENTED** [review]
Nice catch! Looks good, I just have one style comment.

File: satpy/readers/mersi2_l1b.py (hunk 1 above)
Comment (ninahakansson) [reply satpy/readers/mersi2_l1b.py]: Ok

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**